### PR TITLE
fix(controls): stop event propagation from controls

### PR DESCRIPTION
**Pull request by mickr**
* Stop event propagation from preview controls, this is interfering with annotations.
* Add new tests for Controls.js constructor

### CODE REVIEW

File: src/lib/Controls.js [hunk 1]
```diff
@@ -175,9 +175,15 @@ class Controls {
      * @param {Event} event - A DOM-normalized event object.
      * @return {void}
      */
-    clickHandler = () => {
+    clickHandler = event => {
+        if (event) {
+            event.stopPropagation();
+        }
+
+        if (this.hasTouch) {
```
Comment (jstoffan) [inline src/lib/Controls.js:183]: Do we want to hide the controls only for touch-enabled devices?

File: src/lib/Controls.js [hunk 2]
```diff
@@ -175,9 +175,15 @@ class Controls {
...
+            this.shouldHide = true;
+        }
         // If we are not focused in on the page num input, allow hiding after timeout
```
Comment (jstoffan) [inline src/lib/Controls.js:186]: Comment seems to have been orphaned.

File: src/lib/Controls.js (hunk 1 above)
Comment (mickr) [reply src/lib/Controls.js:183]: Yes, I believe that was the intention of the original code as it didn’t register the click handler unless it was a touch device. Now that we are using it, probably should keep the same intention, while only hiding on mobile devices.

File: src/lib/Controls.js [hunk 3]
```diff
@@ -44,10 +44,10 @@ class Controls {
         this.controlsEl.addEventListener('mouseleave', this.mouseleaveHandler);
         this.controlsEl.addEventListener('focusin', this.focusinHandler);
         this.controlsEl.addEventListener('focusout', this.focusoutHandler);
+        this.controlsEl.addEventListener('click', this.clickHandler);
```
Comment (jstoffan) [inline src/lib/Controls.js:47]: Should we move the associated `removeEventListener` call out of its conditional check as well?

Comment (jstoffan) [reply src/lib/Controls.js:47]: Also, this component is used in all of the viewers across Preview, so this is a wide-ranging change. Does it have any side effects? Have we tested all the various viewers?

**Pull request merged**